src/legacy/core_plugins/kibana/public/dashboard/migrations/migrate_to_730_panels.ts

```diff
@@ -0,0 +1,304 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import _ from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_PANEL_WIDTH, DEFAULT_PANEL_HEIGHT } from '../dashboard_constants';
+import { PanelUtils } from '../panel/panel_utils';
+import {
+  RawSavedDashboardPanelTo60,
+  RawSavedDashboardPanel630,
+  RawSavedDashboardPanel640To720,
+  RawSavedDashboardPanel730ToLatest,
+  RawSavedDashboardPanel610,
+  RawSavedDashboardPanel620,
+} from './types';
+import {
+  SavedDashboardPanelTo60,
+  SavedDashboardPanel620,
+  SavedDashboardPanel630,
+  SavedDashboardPanel610,
+} from '../types';
+
+const PANEL_HEIGHT_SCALE_FACTOR = 5;
+const PANEL_HEIGHT_SCALE_FACTOR_WITH_MARGINS = 4;
+const PANEL_WIDTH_SCALE_FACTOR = 4;
+
+/**
+ * Note!
+ *
+ * The 7.3.0 migrations reference versions that are quite old because for a long time all of this
+ * migration logic was done ad hoc in the code itself, not on the indexed data (migrations didn't even
+ * exist at the point most of that logic was put in place).
+ *
+ * So you could have a dashboard in 7.2.0 that was created in 6.3 and it will have data of a different
+ * shape than some other dashboards that were created more recently.
+ *
+ * Moving forward migrations should be simpler since all 7.3.0+ dashboards should finally have the
+ * same data.
+ */
+
+function isPre61Panel(
+  panel: unknown | RawSavedDashboardPanelTo60
+): panel is RawSavedDashboardPanelTo60 {
+  return (panel as RawSavedDashboardPanelTo60).row !== undefined;
+}
+
+function is61Panel(panel: unknown | RawSavedDashboardPanel610): panel is RawSavedDashboardPanel610 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel610).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 1;
+}
+
+function is62Panel(panel: unknown | RawSavedDashboardPanel620): panel is RawSavedDashboardPanel620 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel620).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 2;
+}
+
+function is63Panel(panel: unknown | RawSavedDashboardPanel630): panel is RawSavedDashboardPanel630 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel630).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 3;
+}
+
+function is640To720Panel(
+  panel: unknown | RawSavedDashboardPanel640To720
+): panel is RawSavedDashboardPanel640To720 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel640To720).version);
```

Comment:
For most of the functions in this file, would it be better to use the semver library which we already have in Kibana?

`return semver.intersects(panelVersion, '6.3.0 - 7.3.0')`